exporters/zipkin/src/test/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporterEndToEndHttpTest.java

```diff
@@ -0,0 +1,187 @@
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportAsJsonV1() {
+    ZipkinExporterConfiguration configuration =
+        buildZipkinExporterConfiguration(
+            zipkin.httpUrl() + ENDPOINT_V1_SPANS, Encoding.JSON, SpanBytesEncoder.JSON_V1);
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportFailedAsWrongEncoderUsed() {
+
+    ZipkinExporterConfiguration configuration =
+        buildZipkinExporterConfiguration(
+            zipkin.httpUrl() + ENDPOINT_V2_SPANS, Encoding.JSON, SpanBytesEncoder.PROTO3);
+
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.FAILURE);
+    List<Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans).isEmpty();
+  }
+
+  private static ZipkinExporterConfiguration buildZipkinExporterConfiguration(
+      String endpoint, Encoding encoding, SpanBytesEncoder encoder) {
+    return ZipkinExporterConfiguration.builder()
+        .setSender(URLConnectionSender.newBuilder().endpoint(endpoint).encoding(encoding).build())
+        .setServiceName(SERVICE_NAME)
+        .setEncoder(encoder)
+        .build();
+  }
+
+  /**
+   * Exports a span, verify that it was received by Zipkin, and check that the span stored by Zipkin
+   * matches what was sent.
+   */
+  private void exportAndVerify(ZipkinExporterConfiguration configuration) {
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.SUCCESS);
+    List<Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans.size()).isEqualTo(1);
+    assertThat(zipkinSpans.get(0)).isEqualTo(buildZipkinSpan());
+  }
+
+  private static SpanData.Builder buildStandardSpan() {
+    return SpanData.newBuilder()
+        .setTraceId(TraceId.fromLowerBase16(TRACE_ID, 0))
+        .setSpanId(SpanId.fromLowerBase16(SPAN_ID, 0))
+        .setParentSpanId(SpanId.fromLowerBase16(PARENT_SPAN_ID, 0))
+        .setTraceFlags(TraceFlags.builder().setIsSampled(true).build())
+        .setStatus(Status.OK)
+        .setKind(Kind.SERVER)
+        .setHasRemoteParent(true)
+        .setName("Recv.helloworld.Greeter.SayHello")
+        .setStartEpochNanos(1505855794_194009601L)
+        .setAttributes(attributes)
+        .setTotalAttributeCount(attributes.size())
+        .setTimedEvents(annotations)
+        .setLinks(Collections.<SpanData.Link>emptyList())
+        .setEndEpochNanos(1505855799_465726528L)
```

Comment:
for the purpose of "find", etc either make this a constant or use the same form as below? `1505855799000000L + 465726528L` using the same value representation helps see what's different in the formats

Reply:
thanks, i've extracted these into constants now, should be good to go :)